### PR TITLE
Fix VPC related issues and minor lint updates

### LINKED ISSUE #615: VPC: Invalid zone eu-de-{1,2,3} on cluster creation
/kind bug
/area provider/ibmcloud

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
Setting my API Key and the following endpoints during management cluster setup:
```
export IAM_ENDPOINT=https://iam.cloud.ibm.com/identity/token
export SERVICE_ENDPOINT=https://eu-de.iaas.cloud.ibm.com/v1
```

Settings before provisioning downstream cluster:
```
export IBMVPC_REGION=eu-de
export IBMVPC_ZONE=eu-de-1
export IBMVPC_RESOURCEGROUP=<my-resource-group>
export IBMVPC_NAME=cluster-api
export IBMVPC_IMAGE_ID=<my-capi-ḱ8s-image-id>
export IBMVPC_PROFILE=bx2-4x16
export IBMVPC_SSHKEY_ID=<my-ssh-key-id>
```
Used image: ubuntu-2004-ibmcloud-kube-v1-23-4

```
clusterctl generate cluster ibm-capi-cluster --kubernetes-version v1.23.4 \
--target-namespace cluster-api-ibm \
--control-plane-machine-count=1 \
--worker-machine-count=1 \
--from ./templates/cluster-template.yaml > cluster-start.yaml
```
kubectl apply -f cluster-start.yaml

Logs:

```
I0401 11:25:38.822842       1 logr.go:249] ibmvpcmachine-resource "msg"="default"  "name"="ibm-capi-cluster-md-0-4s8sd"
I0401 11:25:38.943006       1 logr.go:249] ibmvpcmachine-resource "msg"="validate create"  "name"="ibm-capi-cluster-md-0-4s8sd"
I0401 11:25:38.956686       1 ibmvpcmachine_controller.go:77] controllers/IBMVPCMachine "msg"="Machine Controller has not yet set OwnerRef" "ibmvpcmachine"={"Namespace":"cluster-api-ibm","Name":"ibm-capi-cluster-md-0-4s8sd"} 
I0401 11:25:39.222473       1 logr.go:249] ibmvpcmachine-resource "msg"="default"  "name"="ibm-capi-cluster-md-0-4s8sd"
I0401 11:25:39.271179       1 logr.go:249] ibmvpcmachine-resource "msg"="validate update"  "name"="ibm-capi-cluster-md-0-4s8sd"
I0401 11:25:39.608080       1 ibmvpcmachine_controller.go:149] controllers/IBMVPCMachine "msg"="Bootstrap data secret reference is not yet available" "cluster"="ibm-capi-cluster" "ibmvpcmachine"={"Namespace":"cluster-api-ibm","Name":"ibm-capi-cluster-md-0-4s8sd"} 
I0401 11:25:39.722977       1 logr.go:249] ibmvpcmachine-resource "msg"="default"  "name"="ibm-capi-cluster-md-0-4s8sd"
I0401 11:25:39.823175       1 logr.go:249] ibmvpcmachine-resource "msg"="validate update"  "name"="ibm-capi-cluster-md-0-4s8sd"
I0401 11:25:39.864637       1 ibmvpcmachine_controller.go:149] controllers/IBMVPCMachine "msg"="Bootstrap data secret reference is not yet available" "cluster"="ibm-capi-cluster" "ibmvpcmachine"={"Namespace":"cluster-api-ibm","Name":"ibm-capi-cluster-md-0-4s8sd"} 
I0401 11:26:06.843093       1 logr.go:249] ibmvpccluster-resource "msg"="default"  "name"="ibm-capi-cluster"
I0401 11:26:06.982746       1 logr.go:249] ibmvpccluster-resource "msg"="validate update"  "name"="ibm-capi-cluster"
I0401 11:26:07.122647       1 logr.go:249] ibmvpccluster-resource "msg"="default"  "name"="ibm-capi-cluster"
I0401 11:26:07.153230       1 logr.go:249] ibmvpccluster-resource "msg"="validate update"  "name"="ibm-capi-cluster"
E0401 11:26:10.942821       1 controller.go:317] controller/ibmvpccluster "msg"="Reconciler error" "error"="failed to reconcile Control Plane Endpoint for IBMVPCCluster cluster-api-ibm/ibm-capi-cluster: invalid zone eu-de-1" "name"="ibm-capi-cluster" "namespace"="cluster-api-ibm" "reconciler group"="infrastructure.cluster.x-k8s.io" "reconciler kind"="IBMVPCCluster" 
I0401 11:39:30.613505       1 ibmvpcmachine_controller.go:149] controllers/IBMVPCMachine "msg"="Bootstrap data secret reference is not yet available" "cluster"="ibm-capi-cluster" "ibmvpcmachine"={"Namespace":"cluster-api-ibm","Name":"ibm-capi-cluster-md-0-4s8sd"} 
E0401 11:43:20.344957       1 controller.go:317] controller/ibmvpccluster "msg"="Reconciler error" "error"="failed to reconcile Control Plane Endpoint for IBMVPCCluster cluster-api-ibm/ibm-capi-cluster: invalid zone eu-de-1" "name"="ibm-capi-cluster" "namespace"="cluster-api-ibm" "reconciler group"="infrastructure.cluster.x-k8s.io" "reconciler kind"="IBMVPCCluster"
```
**What did you expect to happen:**
I would expect that the region / zone is recognized properly.
Error occurs on all eu-de-1, eu-de-2, eu-de-3

The VPC cluster-api is already existing and an address prefix is already created. 

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api version: v1.1.3
- Minikube/KIND version: Management Cluster v1.22.6+IKS
- Kubernetes version: (use `kubectl version`): Provisioning 1.23.4
- OS (e.g. from `/etc/os-release`): ubuntu-20-04

### PR DESCRIPTION
**Pull request by Prajyot-Parab**
**What this PR does / why we need it**:
- Fix VPC related issues
  - Fix service endpoint url
  - Add proper resource group ID to function calls
- Minor post-liniting process updates

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #615 

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note
Fix VPC related issues and minor lint updates.
```

### CODE REVIEW

File: config/crd/bases/infrastructure.cluster.x-k8s.io_ibmpowervsclusters.yaml [hunk 1]
```diff
@@ -18,7 +18,7 @@ spec:
   - name: v1alpha4
     schema:
       openAPIV3Schema:
-        description: IBMPowerVSCluster is the Schema for the ibmpowervsclusters API
+        description: IBMPowerVSCluster is the Schema for the ibmpowervsclusters API.
```
Comment (mkumatag) [inline config/crd/bases/infrastructure.cluster.x-k8s.io_ibmpowervsclusters.yaml:21]: wondering how it listing didn't fail before?!

File: controllers/ibmvpccluster_controller.go [hunk 2]
```diff
@@ -73,9 +72,8 @@ func (r *IBMVPCClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		return ctrl.Result{}, nil
 	}
 
-	// Create the scope.
-	// TODO: Will be removed once we find a better way of overriding the service endpoint, generate via spec.
-	svcEndpoint := os.Getenv("SERVICE_ENDPOINT")
```
Comment (mkumatag) [inline controllers/ibmvpccluster_controller.go:78]: there are other occurrences of this `SERVICE_ENDPOINT` in the docs, lets remove them as well.

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: controllers/ibmvpccluster_controller.go (hunk 2 above)
Comment (Prajyot-Parab) [reply controllers/ibmvpccluster_controller.go:78]: Done.

File: config/crd/bases/infrastructure.cluster.x-k8s.io_ibmpowervsclusters.yaml (hunk 1 above)
Comment (Prajyot-Parab) [reply config/crd/bases/infrastructure.cluster.x-k8s.io_ibmpowervsclusters.yaml:21]: The changes were added as part of `make manifests` target, so I am assuming we need to add missing `verify-manifests` target to Makefile.

File: docs/book/src/getting-started.md [hunk 3]
```diff
@@ -43,11 +43,10 @@ it into a management cluster using `clusterctl`.
     ```
     - Additional varibles required for VPC
     
-    The sample `IAM_ENDPOINT` below points to Production and the `SERVICE_ENDPOINT` points to the `us-east` VPC region.
+    The sample `IAM_ENDPOINT` below points to Production.
 
     ```console
     export IAM_ENDPOINT=https://iam.cloud.ibm.com/identity/token
```
Comment (mkumatag) [inline docs/book/src/getting-started.md:49]: Even this is not used anywhere, lets remove this as well.

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: docs/book/src/getting-started.md (hunk 3 above)
Comment (Prajyot-Parab) [reply docs/book/src/getting-started.md]: Done.

File: config/crd/bases/infrastructure.cluster.x-k8s.io_ibmpowervsclusters.yaml (hunk 1 above)
Comment (mkumatag) [reply config/crd/bases/infrastructure.cluster.x-k8s.io_ibmpowervsclusters.yaml:21]: lets create a task for the same.

**Review by mkumatag — APPROVED** [review]
/lgtm
/approve

**Pull request merged**